crates/block-producer/src/store/mod.rs

```diff
@@ -56,15 +55,15 @@ pub trait Store: ApplyBlock {
 
     /// Returns note authentication information for the set of specified notes.
     ///
-    /// If authentication info could for a note does not exist in the store, the note is omitted
+    /// If authentication info for a note does not exist in the store, the note is omitted
     /// from the returned set of notes.
     ///
-    /// TODO: right now this return only Merkle paths per note, but this will need to be updated to
-    /// return full authentication info.
+    /// TODO: right now this returns only Merkle paths per note, but this will need to be updated to
+    ///       return full authentication info.
     async fn get_note_authentication_info(
         &self,
         notes: impl Iterator<Item = &NoteId> + Send,
-    ) -> Result<BTreeMap<NoteId, MerklePath>, NotePathsError>;
+    ) -> Result<BTreeMap<NoteId, NoteInclusionProof>, NotePathsError>;
```

Comment:
Maybe we can use a new struct for the return type here - something like:
```rust
pub struct NoteAuthenticationInfo {
    block_proofs: Vec<BlockInclusionProof>,
    note_proofs: BTreeMap<NoteId, NoteInclusionProof>,
}
```
This can the also be used in the `BlockInputs` struct.

Reply:
I'm busy implementing this; but I'm starting to feel like we should have split this PR up much more.

e.g. add this functionality to the store api, and only handle consuming it in the block producer in a separate PR.

Reply:
I'm fine with splitting this into multiple PRs (I too find that it has become a bit difficult to follow) and doing `store` part in one PR and `block-producer` part in another makes sense to me.